packages/apps/reputation-oracle/server/src/common/guards/hcaptcha.ts

```diff
@@ -0,0 +1,57 @@
+import {
+  Injectable,
+  CanActivate,
+  ExecutionContext,
+  HttpStatus,
+  HttpException,
+  Logger,
+} from '@nestjs/common';
+import { Request } from 'express';
+import { HCaptchaService } from '../../integrations/hcaptcha/hcaptcha.service';
+import { AuthConfigService } from '../config/auth-config.service';
+
+@Injectable()
+export class HCaptchaGuard implements CanActivate {
+  logger = new Logger(HCaptchaGuard.name);
+  constructor(
+    private readonly hCaptchaService: HCaptchaService,
+    private readonly authConfigSerice: AuthConfigService,
+  ) {}
+  public async canActivate(context: ExecutionContext): Promise<boolean> {
+    const request: Request = context.switchToHttp().getRequest();
+
+    const { body } = request;
+    const hCaptchaToken = body['h_captcha_token'];
+
+    // TODO: Remove 27-46 lines once we figure out how to replace human app user
+    if (request.path === '/auth/signin') {
+      const email = body['email'];
+      // Need to validate email because guards being called before any interceptors or pipes
```

Comment:
As discussed offline, no need to validate email here since it will be validated later in route pipes, so comment can be removed